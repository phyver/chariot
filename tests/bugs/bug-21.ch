codata stream('x) where
    | Head : stream('x) -> 'x
    | Tail : stream('x) -> stream('x)

val f { Head = x ; Tail = { Head = y ; Tail = { Head = z ; Tail = zs }}} = z

:set verbose 2
:show f
