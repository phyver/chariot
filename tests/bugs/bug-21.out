>> option verbose is set to 1
>> coinductive type stream was succesfully defined
>> Typing for f successful
>> the definition for f is complete
>> the definition f is provably correct

>> option verbose is set to 2
val
    f : stream('a) → 'a
  | f { Head = x ; Tail = { Head = y ; Tail = { Head = z ; Tail = zs }⁰ }⁰ }⁰ = z
>> 
>> corresponding case and struct form:
>> x₁ → x₁.Tail.Tail.Head

